x-pack/plugins/security_solution/public/flyout/document_details/left/components/add_note.tsx

```diff
@@ -96,7 +132,33 @@ export const AddNote = memo(({ eventId }: AddNewNoteProps) => {
         </EuiComment>
       </EuiCommentList>
       <EuiSpacer />
-      <EuiFlexGroup justifyContent="flexEnd" responsive={false}>
+      <EuiFlexGroup alignItems="center" justifyContent="flexEnd" responsive={false}>
+        <EuiFlexItem grow={false}>
+          <>
+            <EuiCheckbox
+              data-test-subj={ATTACH_TO_TIMELINE_CHECKBOX_TEST_ID}
+              id={timelineCheckBoxId}
+              label={
+                <>
+                  {ATTACH_TO_TIMELINE_CHECKBOX}
+                  <EuiToolTip position="top" content={ATTACH_TO_TIMELINE_INFO}>
+                    <EuiIcon
+                      type="iInCircle"
+                      css={css`
+                        margin-left: 4px;
+                      `}
+                    />
+                  </EuiToolTip>
+                </>
+              }
+              disabled={
+                !isTimelineFlyout || (isTimelineFlyout && activeTimeline.savedObjectId == null)
+              }
+              checked={checked}
+              onChange={(e) => onCheckboxChange(e)}
```

Comment:
pull out to a const?

Reply:
I guess I could. Is the value to be a bit more readable?

Reply:
Just react re-render proofing tbh. When you pass an anonymous function, it creates a new function reference on every re-render, which then causes the `EuiCheckbox` component to re-render when nothing's actually changed. Tbh, it's not a performance problem at all for this low-level component, I just suggest this change as good practice

Reply:
oh sorry I thought for a sec that the comment was 2 lines above for the `disabled`